spark-commons/src/test/scala/za/co/absa/spark/commons/OncePerSparkSessionTest.scala

```diff
@@ -48,4 +47,29 @@ class OncePerSparkSessionTest extends AnyFunSuite with MockitoSugar with SparkTe
     assert(libraryBInitCounter == 2)
   }
 
+  test("should return true if the library is registered successfully and false if not, and if spark session hasn't started"){
+    val anotherSpark: SparkSession =  mock[SparkSession]
+    class UDFLibrary() extends OncePerSparkSession(){
+      override protected def registerBody(spark: SparkSession): Unit = {}
+    }
+
+    val UdfLibrary = new UDFLibrary()
+    assert(UdfLibrary.register(anotherSpark))
+    assert(!UdfLibrary.register(anotherSpark))
+  }
+
+  test("should return false for result provided that spark session has started") {
+
+    val anotherSpark: SparkSession =  mock[SparkSession]
+    class UDFLibrary()(implicit sparkToRegisterTo: SparkSession) extends OncePerSparkSession(sparkToRegisterTo) {
+      // Results will yield false because this will be second attempt of registering after the auto register has taken place
+      val results = this.register(sparkToRegisterTo)
```

Comment:
You shouldn't need to (try to) register here. The registering should happening as part of the `OncePerSparkSession` constructor called.
And then let's call the register outside the class in the test body.